src/test/java/com/google/devtools/build/lib/rules/java/JavaInfoRoundtripTest.java

```diff
@@ -38,9 +39,12 @@ public class JavaInfoRoundtripTest extends BuildViewTestCase {
   @Before
   public void javaInfoToDict() throws Exception {
     mockToolsConfig.create("tools/build_defs/inspect/BUILD");
-    mockToolsConfig.copyTool(
-        TestConstants.BAZEL_REPO_SCRATCH + "tools/build_defs/inspect/struct_to_dict.bzl",
```

Comment:
Removing `TestConstants.BAZEL_REPO_SCRATCH` broke the test internally because it has a different value (`third_party/bazel/` internally vs `../io_bazel` externally)

I found the following change fixes the test failure internally, but not sure if it also works in Bazel:
```
"/" + TestConstants.BAZEL_REPO_SCRATCH + "tools/build_defs/inspect/struct_to_dict.bzl"
```

BTW, do we have to change it from a `data` dependency to a `resources` dedendency?

Reply:
The problem is that the runfiles directory isn't created on Windows, so in order to find the dependency we would need to depend on the runfiles library. Can I have this test depend on `@bazel_tools//tools/java/runfiles`? If that reference to `@bazel_tools` is allowed, that would be better than making this a resource.

Reply:
Yes, I think using `@bazel_tools//tools/java/runfiles` is the right way to go, there is also a corresponding internal runfiles library which should just work.